mlir/utils/jenkins/Jenkinsfile.miopen-test

```diff
@@ -1,65 +1,101 @@
-                sh '''
-                    mkdir -p build && cd build
-                    /usr/local/bin/cmake -G "Unix Makefiles" ../llvm \
-                         -DLLVM_ENABLE_PROJECTS="mlir;lld" \
-                         -DCMAKE_BUILD_TYPE=Release \
-                         -DBUILD_SHARED_LIBS=OFF \
-                         -DLLVM_BUILD_LLVM_DYLIB=OFF \
-                         -DLLVM_ENABLE_TERMINFO=OFF
-                    make -j$(nproc) libMLIRMIOpen
-                '''
+        stage("Carry out test") {
+            // Note: the builds are reused between cases on each node
+            // It may be possible to move build steps out to execute once
+            // but this could cause issues with target-specific code
+            // when we run the non-xdlops tests on a gfx900
+            matrix {
+                agent {
+                    docker {
+                        image 'rocm/mlir:miopen_test'
+                        args '--user "$(id -u):$(id -g)" --device=/dev/kfd --device=/dev/dri --group-add video -u 0'
+                        label "${XDLOPS == '1' ? 'gfx908 && multi_gpu' : 'gfx908 && multi_gpu'}"
+                        // The above could have gfx900 || gfx908 in the latter half
+                    }
+                }
+                axes {
+                    axis {
+                        name 'DTYPE'
+                        values 'fp16', 'fp32'
+                    }
+                    axis {
+                        name 'XDLOPS'
+                        values '1', '0'
+                    }
+                    axis {
+                        name 'LAYOUT'
+                        values 'NCHW', 'NHWC'
+                    }
+                    axis {
+                        name 'DIRECTION'
+                        values '1', '4'
+                    }
+                }
+                stages {
+                    stage('Install parallel (temp until dockerfile updated)') {
+                        steps {
+                            sh 'apt-get update && apt-get install -y --no-install-recommends parallel'
+                        }
+                    }
+                    stage('Environment') {
+                        steps {
+                            echo "Config is ($DTYPE, $LAYOUT, $XDLOPS, dir: $DIRECTION)"
+                            sh 'cat /etc/os-release'
+                            sh '/opt/rocm/bin/rocm-smi'
+                        }
+                    }
+                    stage('Build libMLIRMIOpen') {
+                        steps {
+                            checkout scm
+                            cmakeBuild generator: 'Unix Makefiles',\
+                               buildDir: 'build',\
+                               sourceDir: 'llvm',\
+                               installation: 'InDocker',\
+                               buildType: 'Release',\
+                               cmakeArgs: '''-DLLVM_ENABLE_PROJECTS="mlir;lld"
```

Comment:
Excellent! We should do the same thing with other jenkinsfile too. Could you file a ticket for this?

Reply:
Actually on a second thought, leave the other jenkinsfile as-is for now. It is already a pain to manually maintain so many jenkinsfile and jenkins job. I don't know which is matched to which now. I believe we can refactor those when we consolidate all jobs to a single one.